ast/node.go

```diff
@@ -114,12 +114,12 @@ type (
 
 	DotExpression struct {
 		Left       Expression
-		Identifier Identifier
+		Identifier *Identifier
```

Comment:
What is the reason for this change (and the other changes to pointer)?

Reply:
> What is the reason for this change (and the other changes to pointer)?

The only reason for these changes is because their pointer types inherit the Node interface, but the non-pointer type does not (ex: `*Identifier` implements `Node` because its `Idx0` and `Idx1` implementations have pointer receivers, but `Identifier` is not a `Node`). It's basically just to allow easier access to Node types.

Reply:
I don't see a problem, you can always use `&dotExpression.Identifier` if you need to pass a Node somewhere, and accessing methods will just work (e.g. `dotExpression.Identifier.Idx1()`). On the other hand, changing the fields to pointers results in extra allocation and increases memory fragmentation.

Reply:
Good point. I've reverted those changes in the latest commit.